app/scripts/controllers/journalUpload.js

```diff
@@ -0,0 +1,241 @@
+import registrationsPaidPopoverTemplate from 'views/components/registrationsPaidPopover.html';
+import paymentsModalTemplate from 'views/modals/paymentsModal.html';
+import journalUploadReviewModalTemplate from 'views/modals/journalUploadReview.html';
+
+angular
+  .module('confRegistrationWebApp')
+  .controller('journalUploadCtrl', function(
+    $rootScope,
+    $scope,
+    $http,
+    $window,
+    $cookies,
+    $uibModal,
+    registrations,
+    reports,
+    journalUploadService,
+    conference,
+    envService,
+    permissions,
+    modalMessage,
+  ) {
+    $rootScope.globalPage = {
+      type: 'admin',
+      mainClass: 'journal-upload',
+      bodyClass: '',
+      confId: conference.id,
+      footer: true,
+    };
+    $scope.paidPopoverTemplateUrl = registrationsPaidPopoverTemplate;
+    $scope.accountTransfers = journalUploadService.getAccountTransferData(
+      registrations,
+    );
+    $scope.accountTransfersWithErrors = journalUploadService.getAccountTransferDataWithErrors(
+      registrations,
+    );
+    $scope.registrations = registrations;
+    $scope.reports = reports;
+    $scope.currentReportId = null;
+    $scope.conference = conference;
+    $scope.accountTransfersToInclude = [];
+    $scope.queryParameters = {
+      page: 1,
+      limit: 20,
+      orderBy: 'last_name',
+      order: 'ASC',
+      filter: '',
+      filterAccountTransferErrors: 'yes',
+      filterAccountTransfersByExpenseType: '',
+      filterAccountTransfersByPaymentType: '',
+      filterPayment: '',
+      filterRegType: '',
+      includeAccountTransfers: true,
+      includeCheckedin: 'yes',
+      includeWithdrawn: 'yes',
+      includeIncomplete: 'yes',
+    };
+    $scope.meta = {
+      totalPages: 0,
+    };
+    $scope.apiUrl = envService.read('apiUrl');
+    $scope.authToken = $cookies.get('crsToken');
+
+    $scope.$watch(
+      'queryParameters',
+      (newParameters, oldParameters) => {
+        //limit changed, reset page
+        if (
+          newParameters.page > 1 &&
+          newParameters.page === oldParameters.page
+        ) {
+          $scope.queryParameters.page = 1;
+          return;
+        }
+
+        if (newParameters.page !== oldParameters.page) {
+          //scroll to top on page change
+          $window.scrollTo(0, 0);
+        }
+
+        $scope.refresh();
+      },
+      true,
+    );
+
+    $scope.$watch('currentReportId', (newReportId, oldReportId) => {
+      // scroll to top of page if selected report changes
+      if (newReportId !== oldReportId) {
+        $window.scrollTo(0, 0);
+      }
+      // if currentReportId is null, refresh account transfer data
+      if (!newReportId) {
+        $scope.refresh();
+      } else {
+        // set accountTransfers to current report's valid account transfers
+        $scope.accountTransfers = $scope.reports
+          .filter(report => report.id === newReportId)[0]
```

Comment:
Can this find multiple?  Seems like `.find` would be better.
```suggestion
          .filter(report => report.id === newReportId)[0]
```

Also maybe part of the code could be shared with the error one below.

Reply:
I changed it to ```.find``` and was like "Why didn't I do this the first time?" and then realized when I originally used it, it breaks all the tests 🙁 I could use the lodash ```_.find``` but I'm not sure if that's any better.